discovery-provider/src/wsgi.py

```diff
@@ -1,9 +1,21 @@
 # exposes flask create_app() function as gunicorn entrypoint
 import logging
+from os import getenv
 
 from src.app import create_app
 
 logger = logging.getLogger(__name__)
 
+if not getenv("PROMETHEUS_MULTIPROC_DIR"):
+    logger.error("PROMETHEUS_MULTIPROC_DIR is not set")
+    exit(1)
+
+
+def child_exit(server, worker):
```

Comment:
why do we need this function?

Reply:
Added a comment above it after reading into the function call more:

```
# Do bookkeeping when one process dies in a multi-process setup by
# deleting all gauge_${pid}.db files
```